theano/tensor/nlinalg.py

```diff
@@ -179,6 +185,10 @@ class ExtractDiag(Op):
     __props__ = ("view",)
 
     def __init__(self, view=False):
+        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
```

Comment:
The perform code work on ndarray and CudaNDarray object.
Le 8 mars 2016 21:24, "Xavier Bouthillier" <notifications@github.com> a
écrit :

> In theano/tensor/nlinalg.py
> <https://github.com/Theano/Theano/pull/4169#discussion_r55464192>:
>
> > @@ -179,6 +185,10 @@ class ExtractDiag(Op):
> >      __props__ = ("view",)
> >
> >      def __init__(self, view=False):
> > +        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
>
> Yes I saw it and that is what confuses me. How could it run on the GPU
> with only CPU code?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4169/files#r55464192>.
>